containers/src/Kokkos_ScatterView.hpp

```diff
@@ -199,35 +211,30 @@ struct ScatterValue;
 
 template <typename ValueType, typename DeviceType>
 struct ScatterValue<ValueType, Kokkos::Experimental::ScatterSum, DeviceType,
-                    Kokkos::Experimental::ScatterNonAtomic>
-    : Sum<ValueType, DeviceType> {
+                    Kokkos::Experimental::ScatterNonAtomic> {
+  ValueType& value;
+
  public:
   KOKKOS_FORCEINLINE_FUNCTION ScatterValue(ValueType& value_in)
-      : Sum<ValueType, DeviceType>(value_in) {}
+      : value(value_in) {}
   KOKKOS_FORCEINLINE_FUNCTION ScatterValue(ScatterValue&& other)
-      : Sum<ValueType, DeviceType>(other.reference()) {}
+      : value(other.value) {}
   KOKKOS_FORCEINLINE_FUNCTION void operator+=(ValueType const& rhs) {
-    this->join(this->reference(), rhs);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION void operator++() {
-    this->join(this->reference(), 1);
-  }
-  KOKKOS_FORCEINLINE_FUNCTION void operator++(int) {
-    this->join(this->reference(), 1);
+    update(rhs);
   }
+  KOKKOS_FORCEINLINE_FUNCTION void operator++() { update(1); }
+  KOKKOS_FORCEINLINE_FUNCTION void operator++(int) { update(1); }
   KOKKOS_FORCEINLINE_FUNCTION void operator-=(ValueType const& rhs) {
-    this->join(this->reference(), ValueType(-rhs));
-  }
-  KOKKOS_FORCEINLINE_FUNCTION void operator--() {
-    this->join(this->reference(), ValueType(-1));
-  }
-  KOKKOS_FORCEINLINE_FUNCTION void operator--(int) {
-    this->join(this->reference(), ValueType(-1));
+    update(ValueType(-rhs));
   }
+  KOKKOS_FORCEINLINE_FUNCTION void operator--() { update(ValueType(-1)); }
```

Comment:
You probably should have a reason for not using `{}` initialization with numeric literals to avoid accidental narrowing.

Reply:
Wait, what if `ValueType` is unsigned?  It should still be OK, right?

Reply:
> Wait, what if `ValueType` is unsigned? It should still be OK, right?

It will do the right thing